website/client/js/controllers/groupsCtrl.js

```diff
@@ -71,6 +71,35 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       }
     };
 
+    $scope.exportAllMessages = function(messages, timezoneOffset) {
+      
+      if(confirm(env.t('exportInboxMessages'))) {
+
+        var arr = Object.keys(messages).map(function (key) {return messages[key]});
+        
+        // Sort array in descending order
+        arr = _.sortBy(arr, function(num) {
+          return num.sort * -1;
+        });
+
+        var messages = '<html><head></head><body>';
+          for (var i = 0; i < arr.length; i++) {
+            messages += '<p><strong>' + arr[i].user + '</strong> ' + 
+              moment.utc(arr[i].timestamp).zone(timezoneOffset).format('MM/DD/YYYY HH:mm:ss') + 
+              ' (' + (i + 1) + '/' + arr.length + ')<br />' + arr[i].text + '</p>';     
+          }
+          
+        messages += '</body></html>';
+        
+        var blob = new Blob([messages], {type: "text/html"});
```

Comment:
Not sure we support IE9

Il 13 set 2016 16:02, "Blade Barringer" <notifications@github.com> ha
scritto:

> In website/client/js/controllers/groupsCtrl.js
> <https://github.com/HabitRPG/habitrpg/pull/7999#discussion_r78564691>:
>
> > +
> > +        // Sort array in descending order
> > +        arr = _.sortBy(arr, function(num) {
> > +          return num.sort * -1;
> > +        });
> > +
> > +        var messages = '<html><head></head><body>';
> > +          for (var i = 0; i < arr.length; i++) {
> > +            messages += '<p><strong>' + arr[i].user + '</strong> ' +
> > +              moment.utc(arr[i].timestamp).zone(timezoneOffset).format('MM/DD/YYYY HH:mm:ss') +
> > +              ' (' + (i + 1) + '/' + arr.length + ')<br />' + arr[i].text + '</p>';
> > +          }
> > +
> > +        messages += '</body></html>';
> > +
> > +        var blob = new Blob([messages], {type: "text/html"});
>
> We can't really use this way of constructing it as Blob isn't supported
> in IE9 and doesn't really have any support in mobile browsers.
> https://developer.mozilla.org/en-US/docs/Web/API/Blob
>
> We may drop support for IE9 soon, but as of right now, we support it.
>
> Also, doing this entirely on the client means that no other integrations
> can export the inbox messages without creating their own logic. If the API
> handles creating it, we don't have to worry about either issue I've brought
> up.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HabitRPG/habitrpg/pull/7999/files/ba90fbfe031942760f62fc5bedbd8a345ba341a3#r78564691>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAkAV4VsEOHdEeu20eGpc56MEkiVkKMWks5qpqz8gaJpZM4J6gfD>
> .
>